bcache/bcache.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package bcache provides access to statistics exposed by the bcache (Linux
+// block cache).
+package bcache
+
+// Stats contains bcache runtime statistics, parsed from /sys/fs/bcache/.
+//
+// The names and meanings of each statistic were taken from bcache.txt and
+// files in drivers/md/bcache in the Linux kernel source. Counters are float64
+// (in-kernel counters are mostly unsigned long).
+type Stats struct {
+	// The name of the bcache used to source these statistics.
+	Name   string
+	Bcache BcacheStats
+	Bdevs  []BdevStats
+	Caches []CacheStats
+}
+
+// BcacheStats contains statistics tied to a bcache ID.
+type BcacheStats struct {
+	AverageKeySize        float64
```

Comment:
The source does return floats for any number > 1023. However, they are ints turned into "humanized" strings such as "1.0k". Yes, we could convert these numbers into int64 or uint64 before passing them to the caller. I went with float64 mostly for safety reasons; float64 will automatically do the correct thing where uint64 and int64 would fail. The Linux function that does the printing for most of the files contains prefixes up to Yotta which won't fit into a uint64, but it's unlikely to ever occur, especially considering that the input type for said function is int64_t. Also, I have not audited all return values for signed values, so some might conceivably return negative values (although none of them look like they should). I'll take a look at using uint64 instead of float64.